yambo-back/controllers/cardController.js

```diff
@@ -0,0 +1,104 @@
+const Deck = require("../models/deckModel");
+const Card = require("../models/cardModel");
+
+const batchCardCreate = async (cardArray, deckId, userId) => {
+  let cardIdArray = [];
+  for (let i = 0; i < cardArray.length; i++) {
+    const newCard = await new Card({
+      front: cardArray[i].front,
+      back: cardArray[i].back,
+      user: userId,
+      deck: deckId,
+    });
+    await newCard.save();
+    cardIdArray.push(newCard._id);
+  }
+  return cardIdArray;
+};
+
+const cardCreate = async (req, res) => {
+  const { deckId, card } = req.body;
+  const userId = req.user._id;
+  try {
+    const newCard = await new Card({
+      front: card.front,
+      back: card.back,
+      user: userId,
+      deck: deckId,
+    });
+    if (!newCard) {
+      return res.status(409).send({ error: "Error making card" });
+    }
+    await newCard.save();
+
+    const deck = await Deck.findOne({ deckId });
+    await deck.cards.push(newCard);
+    await deck.save();
+
+    res.status(201).send(deck);
+  } catch (e) {
+    res.status(400).send({ error: "Error creating card" });
+  }
+};
+
+const cardCreateBatch = async (req, res) => {
+  const { deckId, cards } = req.body;
+  const userId = req.user._id;
+  try {
+    const newCards = await batchCardCreate(cards, deckId, userId);
+    if (!newCards) {
+      return res.status(409).send({ error: "Error making cards" });
+    }
+    await Deck.findOneAndUpdate(
+      { _id: deckId },
+      { $push: { cards: newCards } }
+    );
+    const foundDeck = await Deck.findById(deckId);
```

Comment:
Not that it will make much difference, but could you just have MongoDB return the updated item using an extra option? ( I wanted to try this before recommending it, but I want to merge this commit: {new: true})

Reply:
It should return the updated deck since the method is finding the same deck that was previously updated--the findOneAndUpdate() method saves the changes, then the findbyId() on line 56 is referencing the same deck. I'll build out the automated tests and run a check if the created card(s) are found in the deck.